modules/minimint.nix

```diff
@@ -0,0 +1,125 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  options.services.minimint = {
+      enable = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Enable Minimint,is a federated Chaumian e-cash mint backed 
+        by bitcoin with deposits and withdrawals that can occur on-chain
+        or via Lightning.
+      '';
+    }; 
+    address = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = "Address to listen for RPC connections.";
+    };
+    port = mkOption {
+      type = types.port;
+      default = 5001;
+      description = "Port to listen for RPC connections.";
+    };
+    extraArgs = mkOption {
+      type = types.separatedString " ";
+      default = "";
+      description = "Extra command line arguments passed to electrs.";
+    };
+    dataDir = mkOption {
+      type = types.path;
+      default = "/var/lib/minimint";
+      description = "The data directory for minimint.";
+    };
+    user = mkOption {
+      type = types.str;
+      default = "minimint";
+      description = "The user as which to run minimint.";
+    };
+    group = mkOption {
+      type = types.str;
+      default = cfg.user;
+      description = "The group as which to run minimint.";
+    };
+    tor.enforce = nbLib.tor.enforce;
+    nodes = {
+      clightning = {
+        enable = mkOption {
+          type = types.bool;
+          default = true;
+          description = "Enable the clightning node interface.";
+        };  
+      };
+    };  
+    package = mkOption {
+      type = types.package;
+      default = config.nix-bitcoin.pkgs.minimint;
+      defaultText = "config.nix-bitcoin.pkgs.minimint";
+      description = "The package providing minimint binaries.";
+    };
+  };
+
+  cfg = config.services.minimint;
+  nbLib = config.nix-bitcoin.lib;
+  nbPkgs = config.nix-bitcoin.pkgs;
+  runAsUser = config.nix-bitcoin.runAsUserCmd;
+  secretsDir = config.nix-bitcoin.secretsDir;
+  bitcoind = config.services.bitcoind;
+
+in {
+  inherit options;
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+    services.bitcoind = {
+      enable = true;
+      txindex = true;
+    };
+    services.clightning.enable = true;
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}' 0770 ${cfg.user} ${cfg.group} - -"
+    ];
+    systemd.services.minimint = {
+      wantedBy = [ "multi-user.target" ];
+      requires = [ "bitcoind.service" "fedimint-gateway.service" ];
+      after = [ "bitcoind.service" "fedimint-gateway.service"  ];
+      preStart = ''
+        echo "auth = \"${bitcoind.rpc.users.public.name}:$(cat ${secretsDir}/bitcoin-rpcpassword-public)\"" \
+          > federation.json
+      '';
+      serviceConfig = nbLib.defaultHardening // {
+      WorkingDirectory = cfg.dataDir;
+      ExecStart = ''
+       
+        fm_bin=$1
+        fm_cfg=$2
+        btc_rpc_address="127.0.0.1:8333"
```

Comment:
Can we get the RPC port from the config? This is the mainnet port and we're not going to use mainnet for some time. At this point we should be able to use regtest or signet.